### DISCUSSION TITLE
Getting confidence intervals for MLE after fitting the model

### DISCUSSION THREAD

**Discussion by ozanyesil** [georgebv/pyextremes#49] [Q&A] [answered]
The estimated parameters can be seen after fitting the model but is there a way to see parameter estimations with upper and lower bounds as in the `get_return_value` method?

**Comment by georgebv** [comment, marked as answer]
You can get all distribution parameters from the models and then calculate confidence bounds for each.

For `MLE` it is `fit_parameter_cache` attribute, you must call the `get_return_value` method with `alpha` argument and with `n_samples` set to the desired sample size https://github.com/georgebv/pyextremes/blob/ca4f3689b1994cadfffc20b1bf5d517f953a6ac1/src/pyextremes/models/model_mle.py#L50

For `Emcee` it is `_trace` attribute which stores chains for each parameter https://github.com/georgebv/pyextremes/blob/ca4f3689b1994cadfffc20b1bf5d517f953a6ac1/src/pyextremes/models/model_emcee.py#L84

For Emcee you can visualize them using the `plot_trace` method https://github.com/georgebv/pyextremes/blob/ca4f3689b1994cadfffc20b1bf5d517f953a6ac1/src/pyextremes/eva.py#L1089-L1126

This is an example from README: https://raw.githubusercontent.com/georgebv/pyextremes-notebooks/master/notebooks/documentation/readme%20figures/trace.png

**Comment by ozanyesil** [comment]
Thank you for your detailed answer, but there are still things that I could not understand. If we think the following code as an example, where can I find parameter estimations or which code I should write? 

model = EVA(data = data)
model.get_extremes(method = 'BM', extremes_type = 'high')
model.fit_model()
model.get_return_value(
    return_period=[1, 2, 5, 10, 25, 50, 100],
    alpha=0.95,
    n_samples=1000,
)

Thank you.

**Reply by georgebv** [reply]
The models I referenced above can be accessed via `model.model`. FYI, by default you fit with MLE.

**Comment by ozanyesil** [comment]
Thanks again for the answer. I have one more question. When I run `model.model` it returns `fit parameter cache size` which I don't know what it means. So, still can not see the bounds of parameter estimations. How should i interpret the results of `model.model`. 
Thank you.

**Reply by georgebv** [reply]
This is an `n` by `m` array with `n` being number of parameters and `m` being number of samples. You should be able to calculate confidence bounds for each of the parameters using this array. You can assume that they follow normal distribution.

**Comment by ozanyesil** [comment]
Alright, thank you for your time and help. I got my answers and closing the discussion.
